UnityProject/Assets/Scripts/UI/CameraZoomHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using UnityEngine;
+
+public class CameraZoomHandler : MonoBehaviour
+{
+	private int zoomLevel = 1;
+	private int cachedZoomLevel = 1;
+
+	void Start()
+	{
+		// Setup initial zoom level by guessing.
+		SetZoomLevel(-1);
+	}
+	
+	// Update is called once per frame
+	void Update()
+	{
+		if(cachedZoomLevel != zoomLevel)
+		{
+			// Calculate ratio.
+			double ratio = Screen.height / (double) Screen.width;
+			// Calculate scaling factor. 409600 is a magic number.
+			double scaleFactor = Math.Sqrt(Screen.height * Screen.width / (409600 * ratio));
+			// Calculate orthographic size with full precision and then convert to float precision.
+			Camera.main.orthographicSize = Convert.ToSingle(ratio * 10 * scaleFactor / zoomLevel);
+			// Recenter camera.
+			DisplayManager.Instance.SetCameraFollowPos();
+			cachedZoomLevel = zoomLevel;
+		}
+	}
+
+	public void SetZoomLevel(float zoomLevel)
```

Comment:
Looks like this never gets triggered on resolution change, why not make this method only change this.zoomLevel, and Refresh() afterwards?